products/appengine/terraform.yaml

```diff
@@ -57,8 +57,9 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         ignore_read: true
       threadsafe: !ruby/object:Overrides::Terraform::PropertyOverride
         ignore_read: true
+      # instanceClass defaults to a value based on the scaling method
       instanceClass: !ruby/object:Overrides::Terraform::PropertyOverride
-        ignore_read: true    
+        default_from_api: true
```

Comment:
keeping the `ignore_read` was not an option as that failed the integration test. But since the field itself is optional and the default value depends om the type of scaling, I decided to use `default_from_api`. I chose that based on other fields and and assumption on what this does.
But I am a little outside of my comfort zone here, so please check if this is the correct approach

Reply:
Hmmm, I'd be curious as to why `ignore_read` causes the test to fail. Does it happen during the update step when you modify the instance class? What does the failure look like?

Generally default_from_api does make sense in this case, as it marks the field as `Computed` which means it can take its value from the API version if it is not specified.

Reply:
I'm not 100% sure what is happening with `ignore_read`, but it seems like it is not reading back the instanceClass property so it determines it to be empty, which fails the test as the test expects it to be F2:
```
% make testacc TEST=./google TESTARGS='-run=TestAccAppEngineStandardAppVersion_update'
==> Checking source code against gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google -v -run=TestAccAppEngineStandardAppVersion_update -timeout 240m -ldflags="-X=github.com/terraform-providers/terraform-provider-google/version.ProviderVersion=acc"
=== RUN   TestAccAppEngineStandardAppVersion_update
=== PAUSE TestAccAppEngineStandardAppVersion_update
=== CONT  TestAccAppEngineStandardAppVersion_update
    TestAccAppEngineStandardAppVersion_update: testing.go:635: Step 1 error: ImportStateVerify attributes not equivalent. Difference is shown below. Top is actual, bottom is expected.

        (map[string]string) {
        }


        (map[string]string) (len=1) {
         (string) (len=14) "instance_class": (string) (len=2) "F2"
        }

--- FAIL: TestAccAppEngineStandardAppVersion_update (190.12s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-google/google	190.634s
FAIL
make: *** [testacc] Error 1
```

Reply:
I also figures that `default_from_api` sounds like a less brute force method than simply `ignore_read` making `default_from_api` preferable. But that is just my gut feeling based on my own expectations what these options mean.

Reply:
Ah, that actually makes sense, and `ignore_read` was probably incorrect in the first place.

The step that the test is failing on is an import test, which imports the resource from the existing resource in GCP. Because `instance_class` is ignore_read, it cannot fetch this information from GCP and fails because the imported state doesn't match the previous state.

There was no way for this to work during import before